test/mirror_vfd.c

```diff
@@ -2560,6 +2581,111 @@ test_concurrent_access(void)
     return -1;
 } /* end test_concurrent_access() */
 
+/* ----------------------------------------------------------------------------
+ * Function:    parse_args
+ *
+ * Purpose:     Parse command-line arguments, populating the options struct
+ *              pointer as appropriate.
+ *              Default values will be set for unspecified options.
+ *
+ * Return:      0 on success, negative (-1) if error.
+ * ----------------------------------------------------------------------------
+ */
+static int
+parse_args(int argc, char **argv, struct mt_opts *opts)
+{
+    int i = 0;
+
+    opts->portno = SERVER_HANDSHAKE_PORT;
+    HDstrncpy(opts->ip, SERVER_IP_ADDRESS, H5FD_MIRROR_MAX_IP_LEN);
+
+    for (i = 1; i < argc; i++) { /* start with first possible option argument */
+        if (!HDstrncmp(argv[i], "--ip=", 5)) {
+            HDstrncpy(opts->ip, argv[i] + 5, H5FD_MIRROR_MAX_IP_LEN);
+        }
+        else if (!HDstrncmp(argv[i], "--port=", 7)) {
+            opts->portno = HDatoi(argv[i] + 7);
+        }
+        else {
+            HDprintf("Unrecognized option: '%s'\n", argv[i]);
+            return -1;
+        }
+    } /* end for each argument from command line */
+
+    /* auto-replace 'localhost' with numeric IP */
+    if (!HDstrncmp(opts->ip, "localhost", 10)) { /* include null terminator */
+        HDstrncpy(opts->ip, "127.0.0.1", H5FD_MIRROR_MAX_IP_LEN);
+    }
+
+    return 0;
+} /* end parse_args() */
+
+/* ----------------------------------------------------------------------------
+ * Function:    confirm_server
+ *
+ * Purpose:     Create socket and confirm remote server is available.
+ *
+ * Return:      0 on success, negative (-1) if error.
+ * ----------------------------------------------------------------------------
+ */
+static int
+confirm_server(struct mt_opts *opts)
```

Comment:
Wait for the server to be available, since it was started as a background process in the test script.